src/shogun/util/factory.h

```diff
@@ -267,5 +274,21 @@ namespace shogun
 		SG_REF(result);
 		return result;
 	}
+
+	/** Create a pipeline builder.
+	 * See also CPipelineBuilder and CPipeline.
+	 * @return new instance of CPipelineBuilder
+	 */
+	CPipelineBuilder* pipeline()
```

Comment:
@gf712 would you be up for making the definitions of the factory methods a bit neater? :)

Reply:
sure! what do you have in mind?

Reply:
Actually, just checked, not sure `factory.h` can even be made that much simpler?